enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,476 @@
+   `openshift-sdn-node` process.
+
+4. CNO will also deploy OVN Kubernetes to the cluster with hybrid overlay
+   enabled. To avoid racing, the host subnet allocation will be disabled in
+   OVN Kubernetes.
+   
+   The ovnkube-node container wrapper script will be rendered with the logic for
+   migration:
+
+   - If `br-ex` doesn't exist, it means the node has not yet been updated by
+     MCO, so instead of starting the `ovnkube-node` process, it would run the
+     `hybrid-overlay-node` process. `hybrid-overlay-node` can add the node
+     annotation `k8s.ovn.org/hybrid-overlay-distributed-router-gateway-mac`,
+     which is required by hybrid overlay.
+
+     The script will also add the node annotation
+     `k8s.ovn.org/hybrid-overlay-node-subnet`, according to the HostSubnet CR of
+     this node.
+
+   - If `br-ex` does exist, it means the node has been updated by MCO. So the
+     `ovnkube-node` can be started.
+
+     The script will also add the node annotation `k8s.ovn.org/node-subnets`
+     according to the HostSubnet CR of this node.
+
+#### Migration
+
+When the condition 'Progressing' of the network Cluster Operator is switched
+from `True` to `False`, the admin can kick off the migration from CNO by:
+
+```bash
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"spec":{"networkType":"OVNKubernetes"}'
+```
+
+CNO will update the `status.networkType` field of the `network.config` CR. When
+MCO renders the MachineConfig, some host network configuration services will be
+rendered differently for different CNIs. For example, the
+`ovs-configuration.service` will create the br-ex bridge if the
+`status.networkType` is `OVNKubernetes`; it will not otherwise. So the change of
+the `status.networkType` field will trigger MCO to apply a new MachineConfig to
+each node:
+
+1. MCO will 
+   - render the MachineConfig for each node.
+   - try to cordon and drain the nodes based on
+     `MachineConfigPools.spec.maxUnavailable`
+2. MCO will drain and reboot the node.
+3. When the node is up, br-ex will be created. It means the node is ready for
+   OVN Kubernetes. So the local zone OVN Kubernetes components can be started.
+   The ovnkube-node container wrapper script will also add the node annotation
+  `k8s.ovn.org/node-subnets`, according to the HostSubnet CR of this node.
+4. On the local node, OVN Kubernetes will add all SDN nodes as hybrid overlay
+   nodes to the local zone and all OVN nodes as OVN-IC remote zones. On a remote
+   OVN node, OVN Kubernetes will remove a hybrid overlay node and add an OVN-IC
+   remote zone.
+5. MCO will uncordon the node. Pods will be recreated on the node using
+   ovn-kubernetes as the default CNI plugin.
+
+The above process will be repeated for each node until all the nodes have been
+applied to the new MachineConfig and converted to OVN Kubernetes.
+
+#### Migration Cleanup
+
+Once migration is complete, the admin can trigger the migration cleanup by:
```

Comment:
Same here.. we're again stating that a human has to do this, rather than an operator just cleaning up when it's done

Reply:
osdctl is an SRE tool, it isn't used in automation afaik but could be wrong.  It's not the right place to think about automatino for the SD use case.  we cannot have SRE running migrations for anything, it needs to be a controller somewhere, preferably in-cluster.